### PR TITLE
Fix cross-page element dragging in RTL

### LINKED ISSUE #3012: Fix RTL styling in Stories editor
## Bug Description

The Stories editor is currently not usable at all when using an RTL language. Even with the changes in #2977 applied.

It's mostly arrow icons that are not flipped, and the `transform` in the `EditorCarousel` component that doesn't change when `isRTL`.

## Expected Behaviour

The stories editor should work just fine in RTL mode.

## Steps to reproduce

1. Install [RTL Tester](https://wordpress.org/plugins/rtl-tester/) plugin and click on "Switch to RTL" in the admin bar. _Or_, alternatively, switch user language to an RTL language like Arabic.
2. Open stories editor
3. Notice how not all of the layout elements are "flipped" and the editor is not usable
4. Notice that the context menu doesn't seem to be working either.

## Screenshots

Screenshot 2019-08-12 at 15 15 46

## Additional context

 - WordPress version: 5.3
 - Plugin version: current `develop`
 - Gutenberg plugin version (if applicable): 6.6

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* [x] The current page in the carousel should be correctly centered and navigation should be possible the same way as in LTR mode
* [x] Iconography should be adapted (flipped) accordingly in the context menu as well as the carousel navigation
* ~[ ] It should be possible to access the context menu as expected~ -> see #3607
* [x] Dragging between pages should work as expected and highlight the correct target pages
* [x] Resize handles should be displayed the same way as in LTR mode

## Implementation brief

* Wherever code changes are needed, `select( 'core/block-editor' ).getSettings().isRTL` is used to switch the logic for RTL mode.
* If necessary, CSS changes are made to improve RTL compatibility.
* Add `rtl:ignore` comments for resize handles

### Setup before any test below
1. Activate RTL Tester plugin
1. Click on "Switch to RTL" in the admin toolbar at the top
1. Create a new story

### Verify carousel
1. Create few pages and go through the carousel
1. **Verify:** the currently selected page should always be shown in the center
1. **Verify:** carousel arrows should point into the right direction and navigate correctly
1. **Verify:** dragging pages in carousel works as intended

### Verify resize handles
1. Edit text block
1. **Verify:** the resize handles should be displayed at every location correctly

### Verify dragging across pages
1. Drag block to another page
1. **Verify:** the correct page is highlighted when hovering over it
1. **Verify:** the correct placement when dropped on another place

### Verify CTA block layout
1. Add CTA block on a new page
1. Click on button to edit the URL
1. **Verify:** the URL input field should not overlap the button

## Demo

* https://youtu.be/qfksdQxlFZI

## Changelog entry

* Improved RTL support in the editor

### PR DESCRIPTION
**Pull request by barklund**
As discovered in #3634 block dragging across pages doesn't work in RTL as the moved element is offset by several pages on the new page. This is fixed by changing the direction of the offsetting based off of RTL setting.

_Fixes #3634. Fixes #3012._

## Checklist

- [x] My pull request is addressing an [open issue](https://github.com/ampproject/amp-wp/contributing/project-management.md#life-of-an-issue) (please create one otherwise).
- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/contributing/engineering.md#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/contributing/engineering.md) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: assets/src/stories-editor/components/block-mover/draggable.js [hunk 1]
```diff
@@ -90,7 +98,8 @@ class Draggable extends Component {
 			// All this is about calculating the position of the (correct) element on the new page.
 			const currentElementTop = parseInt( this.cloneWrapper.style.top );
 			const currentElementLeft = parseInt( this.cloneWrapper.style.left );
-			const newLeft = currentElementLeft - ( this.pageOffset * PAGE_AND_MARGIN );
+			const factor = isRTL ? 1 : -1;
+			const newLeft = currentElementLeft + factor * ( this.pageOffset * PAGE_AND_MARGIN );
```
Comment (swissspidy) [inline assets/src/stories-editor/components/block-mover/draggable.js]: > `Unexpected mix of '+' and '*'  no-mixed-operators`

Comment (swissspidy) [reply assets/src/stories-editor/components/block-mover/draggable.js]:
```suggestion
			const newLeft = currentElementLeft + ( factor * ( this.pageOffset * PAGE_AND_MARGIN ) );
```

Comment (barklund) [reply assets/src/stories-editor/components/block-mover/draggable.js]: *Please Excuse My Dear Aunt Sally*

(also, fixing 😜 )

Comment (spacedmonkey) [inline assets/src/stories-editor/components/block-mover/draggable.js]:
```suggestion
			const newLeft = currentElementLeft + ( factor * ( this.pageOffset * PAGE_AND_MARGIN ) );
```

Comment (barklund) [reply assets/src/stories-editor/components/block-mover/draggable.js]: Fixed in a11831f.

**Pull request merged**